src/base/bittorrent/session.cpp

```diff
@@ -3044,13 +3049,18 @@ void Session::startUpTorrents()
         QByteArray data;
     } TorrentResumeData;
 
-    auto startupTorrent = [this, logger, resumeDataDir](const TorrentResumeData &params)
+    int addCount = 0;
+    auto startupTorrent = [this, logger, resumeDataDir, addCount](const TorrentResumeData &params)
     {
         QString filePath = resumeDataDir.filePath(QString("%1.torrent").arg(params.hash));
         qDebug() << "Starting up torrent" << params.hash << "...";
         if (!addTorrent_impl(params.addTorrentData, params.magnetUri, TorrentInfo::loadFromFile(filePath), params.data))
             logger->addMessage(tr("Unable to resume torrent '%1'.", "e.g: Unable to resume torrent 'hash'.")
                                .arg(params.hash), Log::CRITICAL);
+        // process add torrent messages before message queue overflow
+        if(addCount++ % 100 == 0){
```

Comment:
I will change it, I am not used to define by reference manually ;)
I don't know why they decide that by reference it's not default for c++